drivers: uart_nxp_s32_linflexd: support initialization configuration via devicetree

congnguyenhuu opened the pull request:
Added support for initialization configuration via Devicetree.
Enhanced uart-controller.yaml to expand parity type support, now including 'mark' and 'space' options.

congnguyenhuu committed:
drivers: uart_nxp_s32_linflexd: support config via devicetree

Added support for initialization configuration via Devicetree.

Signed-off-by: Cong Nguyen Huu <cong.nguyenhuu@nxp.com>

manuargue commented on drivers/serial/uart_nxp_s32_linflexd.c:
```diff
@@ -362,6 +401,17 @@ static DEVICE_API(uart, uart_nxp_s32_driver_api) = {
 	}
 
 #define UART_NXP_S32_INIT_DEVICE(n)						\
+	BUILD_ASSERT(DT_INST_ENUM_IDX(n, stop_bits) == UART_CFG_STOP_BITS_1 ||	\
```
why not overriding the binding defaults in `nxp,s32-linflexd.yaml`, e.g. allowing `stop-bits` to be `enum: ["1", "2"]`?
Unless there's a dts guideline that says this is not recommended.. but I don't see any in https://docs.zephyrproject.org/latest/build/dts/bindings-upstream.html

congnguyenhuu replied:
The binding does not allow overwriting the default enum values of a property defined in the included binding `uart-controller.yaml`. If try to overwrite , an error will occur:
![image](https://github.com/user-attachments/assets/2d65fa31-e94b-43d9-a791-bddb58c8640c)

Or you are suggesting duplicating the properties already defined in `uart-controller.yaml` and remove include `uart-controller.yaml`?

Other point, I want the value enum binding index must to be define to correspond with the value enum configuration in [uart.h](https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/drivers/uart.h#L87). Due to it's used to convert to the value hw config.

rruuaanng commented on drivers/serial/uart_nxp_s32_linflexd.h:
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2022-2023 NXP
+ * Copyright 2022-2023, 2025 NXP
```
```suggestion
 * Copyright 2022-2025 NXP
```

congnguyenhuu replied:
This copyright applies to continuous commitment years from 2022 to 2025. But this file does not include commitments in 2024.

dleach02 replied:
@congnguyenhuu is correct. The copyright is how you do this if you skipped years of changes.

dcpleung commented:
Please separate the change to `uart-controller.yaml` into another commit.

congnguyenhuu committed:
dts: bindings: serial: uart-controller: expand parity type support

Enhanced uart-controller.yaml to expand parity type support,
now including 'mark' and 'space' options.

Signed-off-by: Cong Nguyen Huu <cong.nguyenhuu@nxp.com>

congnguyenhuu commented:
> Please separate the change to `uart-controller.yaml` into another commit.

I separated a commit for this change
